pkg/manager/impl/execution_manager.go

```diff
@@ -213,24 +215,42 @@ func createTaskDefaultLimits(ctx context.Context, task *core.CompiledTask) runti
 	return runtimeInterfaces.TaskResourceSet{CPU: cpuLimit, Memory: memoryLimit}
 }
 
+// This type stores the locations of where a CPU and memory resource are defined in a []core.Resources_ResourceEntry.
+type taskResourceIndices struct {
```

Comment:
Could you say a bit more why we need these indices?

Reply:
sure the resources are a list (rather than a map) so we need to know where to reference specific resource values by resource type